src/js/builtins/Module.ts

```diff
@@ -34,53 +47,84 @@ export function overridableRequire(this: CommonJSModuleRecord, id: string) {
     $evaluateCommonJSModule(existing);
     return existing.exports;
   }
-
   if (id.endsWith(".node")) {
     return $internalRequire(id);
   }
-
   // To handle import/export cycles, we need to create a module object and put
   // it into the map before we import it.
   const mod = $createCommonJSModule(id, {}, false, this);
   $requireMap.$set(id, mod);
 
+  let { Module } = cache;
+  if (Module === undefined) {
+    // @ts-ignore - TS doesn't know each function is isolated.
+    Module = require("node:module");
+    cache.Module = Module;
+  }
+  const { _extensions } = Module;
+  if (_extensions && _extensions.$structureChanged) {
```

Comment:
Instead of adding a private property, could we do this in `this.$require`?

We could add the StructureID as a field on the Zig::GlobalObject at the time of creation (with some way to mark it as unknown because it was never in use)

Then we can avoid `cache` + these property lookups in JS + pulling in `basename`

For figuring out the file extension from C++, we can do `.findLast('.')` I think on the `WTF::WTFString`

Reply:
Yap. Doable!

Reply:
I tried setting a global value based on the structure of the extensions object but it always being different than the real one. Not sure why creating the object in an initLater and getting its structure was any different than within the require function creation. Setting it on the real extensions object though does seem to work.